ql/instruments/floatfloatswaption.cpp

```diff
@@ -47,12 +49,32 @@ namespace QuantLib {
 
         arguments->swap = swap_;
         arguments->exercise = exercise_;
+        arguments->settlementType = settlementType_;
+        if(settlementMethod_) {
+            arguments->settlementMethod = *settlementMethod_;
+        }
+        else {
+            // if the settlement method is not given, default
+            // values are used
+            arguments->settlementMethod =
+                settlementType_ == Settlement::Physical
+                    ? SettlementMethod::Physical
+                    : SettlementMethod::ParYieldCurve;
+        }
     }
 
     void FloatFloatSwaption::arguments::validate() const {
         FloatFloatSwap::arguments::validate();
         QL_REQUIRE(swap, "underlying cms swap not set");
         QL_REQUIRE(exercise, "exercise not set");
+        QL_REQUIRE(settlementType != Settlement::Physical ||
```

Comment:
...and these two could be a static `Settlement::validate(Type, Method)` method.

Reply:
agreed